### PR TITLE
Added pre-compilation of JSX

### PR DESCRIPTION
**Pull request by andreweduffy**
To avoid having to transform JSX in-browser, added a build step that
will eventually pre-compile all resources (LESS, JSX).

Caveats:

* Just does JSX atm, but can be updated to compile LESS as well.
* Still need the JSXTransformer for inline JSX, would be best to move
  that out to a separate file.
* Would also be nice to perform minification as well.

### CODE REVIEW

**Comment by andreweduffy** [comment]
I've included a README inside of apps/web root, but succinctly:

- Need to install NPM
- run `npm run jsx` from the root of the directory. This step is required before you can serve content
- run `npm run clean` to clean the compiled JSX.

I'd like to add ability to compile LESS as well, also could not eliminate the JSXTransformer completely because of inline JSX in `detail.html`

**Comment by percyliang** [comment]
Cool!

On Tue, Jan 19, 2016 at 2:39 PM, Andrew <notifications@github.com> wrote:

> I've included a README inside of apps/web root, but succinctly:
>
>    - Need to install NPM
>    - run npm run jsx from the root of the directory. This step is
>    required before you can serve content
>    - run npm run clean to clean the compiled JSX.
>
>
Could you also add instructions on how to install gulp, etc.?  Should this
eventually go on the wiki?


>
>
> I'd like to add ability to compile LESS as well, also could not eliminate
> the JSXTransformer completely because of inline JSX in detail.html
>
Is it just {{STATIC_URL}}?  Can we just get rid of this?

**Comment by andreweduffy** [comment]
> Could you also add instructions on how to install gulp, etc.?  Should this
eventually go on the wiki?

You install it when you run `npm i[nstall]`

> Is it just {{STATIC_URL}}?  Can we just get rid of this?

Yeah I can factor out the inline jsx script into a separate file, if that's what you mean

**Comment by andreweduffy** [comment]
@percyliang This should be GTG now

File: codalab/apps/web/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+We have a few compiled resources, namely a set of LESS files as well as JSX that
+we have inside our React components. Compilation is done through [NPM scripts](https://docs.npmjs.com/misc/scripts),
+the relevant ones are below:
+
+* **Compiling JSX**:
+
+        $ npm run jsx
+
+* **Cleaning**
```
Comment (percyliang) [inline codalab/apps/web/README.md]: Say one sentence about when you would run this command.

File: codalab/apps/web/README.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        $ npm i
+
+Building
+=========
+
+We have a few compiled resources, namely a set of LESS files as well as JSX that
+we have inside our React components. Compilation is done through [NPM scripts](https://docs.npmjs.com/misc/scripts),
+the relevant ones are below:
+
+* **Compiling JSX**:
```
Comment (percyliang) [inline codalab/apps/web/README.md]: Say what files are being written out.

File: codalab/apps/web/README.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+Setup
+======
+
+You must install the latest version of [Node.js](https://nodejs.org/en/) for your platform.
+Node is a JavaScript runtime that's used often to write front-end tools. We use Node.js's
+supplied package manager, NPM, to track the dependencies for our front-end build tools.
+
+Once you've installed Node, you'll need to install development dependencies:
+
+        $ cd WORKSHEETS_HOME/codalab/apps/web
```
Comment (percyliang) [inline codalab/apps/web/README.md]: I don't think you need the 'cd'.  Just say 'in this directory'...if they're reading this, then they're probably already in this directory.

**Comment by percyliang** [comment]
One other comment: we should add documentation 
https://github.com/codalab/codalab-worksheets/wiki/Setup-Local-Worksheets
In particular a new section on how to update CodaLab (e.g., ./manage syncdb --migrate and npm run jsx)

Otherwise, LGTM!

**Comment by andreweduffy** [comment]
Sounds good! Updated the README, I'll add to the wiki as well

**Pull request merged**